### PR TITLE
Remind the user to provide a License in Python codegen.

### PR DESCRIPTION
**Pull request by RobbieMcKinstry**
# Description

This PR is similar to #12287; it turns access to the license file into a checked
error instead of an unchecked error, improving safety. It also issues a log
when the field is unset in the schema, since the License field should be set
under most circumstances.

Fixes https://github.com/pulumi/pulumi/issues/12286

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by RobbieMcKinstry** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12287** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/12287" target="_blank">Graphite</a> 
    * **PR #12288** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/12288" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pulumi/pulumi/12288?utm_source=stack-comment).

File: pkg/codegen/python/gen.go [hunk 1]
```diff
@@ -2151,8 +2151,12 @@ func genPackageMetadata(
 		fmt.Fprintf(w, "          'Repository': '%s'\n", pkg.Repository)
 		fmt.Fprintf(w, "      },\n")
 	}
-	if pkg.License != "" {
-		fmt.Fprintf(w, "      license='%s',\n", pkg.License)
+	// Remind the user to provide a license if none is provided in
+	// the schema.
+	if license, err := pkg.GetLicense(); err == nil {
```
Comment (Frassle) [inline pkg/codegen/python/gen.go:2156]: I guess this isn't unique to Python? Should this just be an error when binding a schema?

Comment (RobbieMcKinstry) [reply pkg/codegen/python/gen.go:2156]: Yes, probably!

**Comment by lunaris** [comment]
Closing this as I think adding licence checks to schema binding would introduce a lot of noise for a lot of customers that we might immediately need to allow them to opt out of. Happy to be challenged on this though!

**Pull request closed without merging**